flink/src/test/java/org/apache/iceberg/flink/TestFlinkSchemaUtil.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink;
+
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.Types;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestFlinkSchemaUtil {
+
+  @Test
+  public void testConvertFlinkSchemaToIcebergSchema() {
+    TableSchema flinkSchema = TableSchema.builder()
+        .field("id", DataTypes.INT().notNull())
+        .field("name", DataTypes.STRING()) /* optional by default */
+        .field("salary", DataTypes.DOUBLE().notNull())
+        .field("locations", DataTypes.MAP(DataTypes.STRING(),
+            DataTypes.ROW(DataTypes.FIELD("posX", DataTypes.DOUBLE().notNull(), "X field"),
+                DataTypes.FIELD("posY", DataTypes.DOUBLE().notNull(), "Y field"))))
+        .field("strArray", DataTypes.ARRAY(DataTypes.STRING()).nullable())
+        .field("intArray", DataTypes.ARRAY(DataTypes.INT()).nullable())
+        .field("char", DataTypes.CHAR(10).notNull())
+        .field("varchar", DataTypes.VARCHAR(10).notNull())
+        .field("boolean", DataTypes.BOOLEAN().nullable())
+        .field("tinyint", DataTypes.TINYINT())
+        .field("smallint", DataTypes.SMALLINT())
+        .field("bigint", DataTypes.BIGINT())
+        .field("varbinary", DataTypes.VARBINARY(10))
+        .field("binary", DataTypes.BINARY(10))
+        .field("time", DataTypes.TIME())
+        .field("timestamp", DataTypes.TIMESTAMP())
+        .field("date", DataTypes.DATE())
+        .field("decimal", DataTypes.DECIMAL(2, 2))
```

Comment:
It would be good to use a different precision and scale so that this validates that they are passed correctly, since both are ints.